### PR TITLE
Prepare to rename master branch for main.

### PR DESCRIPTION
**Pull request by ashb**
Part of #12982

There are many more references to "master" (even in our own repo) than
this, but this commit is the first step: to that process.

It makes CI run on the main branch (once it exists), re-words a few
cases where we can to easily not refer to master anymore.

This doesn't yet re-name the `constraints-master` or `master-*` images -
that will be done in a future PR.

(We don't be able to entirely eliminate "master" from our repo as we
refer to a lot of other GitHub repos that we can't change.)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -203,8 +203,8 @@ ENV AIRFLOW_PRE_CACHED_PIP_PACKAGES=${AIRFLOW_PRE_CACHED_PIP_PACKAGES}
 ARG INSTALL_PROVIDERS_FROM_SOURCES="false"
 ENV INSTALL_PROVIDERS_FROM_SOURCES=${INSTALL_PROVIDERS_FROM_SOURCES}
 
-# Only copy install_airflow_from_latest_master.sh to not invalidate cache on other script changes
-COPY scripts/docker/install_airflow_from_latest_master.sh /scripts/docker/install_airflow_from_latest_master.sh
+# Only copy install_airflow_frombranch_tip.sh to not invalidate cache on other script changes
```
Comment (jedcunningham) [inline Dockerfile]: nit: missing underscore, should be `install_airflow_from_branch_tip.sh`

File: Dockerfile.ci [hunk 2]
```diff
@@ -292,8 +292,8 @@ ENV PIP_PROGRESS_BAR=${PIP_PROGRESS_BAR}
 
 RUN pip install --no-cache-dir --upgrade "pip==${AIRFLOW_PIP_VERSION}"
 
-# Only copy install_airflow_from_latest_master.sh to not invalidate cache on other script changes
-COPY scripts/docker/install_airflow_from_latest_master.sh /scripts/docker/install_airflow_from_latest_master.sh
+# Only copy install_airflow_from_latest_main.sh to not invalidate cache on other script changes
```
Comment (jedcunningham) [inline Dockerfile.ci]: nit: should be `install_airflow_from_branch_tip.sh`

Comment (ashb) [reply Dockerfile.ci]: Fixed, thanks

File: Dockerfile (hunk 1 above)
Comment (ashb) [reply Dockerfile]: Fixed, thanks

File: docs/conf.py [hunk 3]
```diff
@@ -76,10 +76,10 @@
     except StopIteration:
         raise Exception(f"Could not find provider.yaml file for package: {PACKAGE_NAME}")
     PACKAGE_DIR = CURRENT_PROVIDER['package-dir']
-    PACKAGE_VERSION = 'master'
+    PACKAGE_VERSION = 'devel'
```
Comment (kaxil) [inline docs/conf.py:79]: Wait are we naming it `devel` or `main` ?

Comment (kaxil) [reply docs/conf.py:79]: It would good to have the same name in the docs as the branch name

Comment (ashb) [reply docs/conf.py:79]: Branch was going to be main, I was thinking that "main" in the docs isn't clear enough though.

It appears we don't actually show this anywhere though, so it's probably moot?

Comment (ashb) [reply docs/conf.py:79]: Can change it if you'd like

Comment (kaxil) [reply docs/conf.py:79]: Yeah since we don't use RTD, it does matter I guess

**Review by kaxil — APPROVED** [review]

**Pull request merged**